pkg/api/v1/schema.go

```diff
@@ -72,12 +62,20 @@ type Disk struct {
 
 type ReadOnly struct{}
 
+// This generally follows style of k8s VolumeSource struct
 type DiskSource struct {
-	File          string          `json:"file,omitempty"`
-	StartupPolicy string          `json:"startupPolicy,omitempty"`
-	Protocol      string          `json:"protocol,omitempty"`
-	Name          string          `json:"name,omitempty"`
-	Host          *DiskSourceHost `json:"host,omitempty"`
+	PersistentVolumeClaim *DiskSourcePersistentVolumeClaim `json:"persistentVolumeClaim,omitempty"`
+	ISCSI                 *DiskSourceISCSI                 `json:"iscsi,omitempty"`
```

Comment:
To keep the code simple I wonder if it would make sense to support PVC only for the moment.

After all we can address iSCSI, ceph rbd, and gluster with them.
The one thing however what we should mention is the way how they should be consumed.

I.e. right now we use qemu's native driver. But under the assumption the we will have  a way to get the kubelet attaching the raw devices for us, then we need a way to signal if we want to use the driver or kubelet way.